### PR TITLE
fix: project environments order

### PR DESCRIPTION
**Pull request by Tymek**
## About the changes
Environments in `/api/admin/projects/$id` are not sorted. Having it in correct order from backend will make new _Single Project Overview - Features list_ table easier to implement.

### Important files
`src/lib/db/project-store.ts`

### CODE REVIEW

File: src/lib/db/project-store.ts [hunk 1]
```diff
@@ -238,7 +238,15 @@ class ProjectStore implements IProjectStore {
...
             })
-            .pluck('environment_name');
+            .innerJoin(
+                'environments',
+                'project_environments.environment_name',
+                'environments.name',
+            )
+            .orderBy('environments.sort_order', 'asc')
+            .orderBy('project_environments.environment_name', 'asc')
+            .pluck('project_environments.environment_name')
```
Comment (sighphyre) [inline src/lib/db/project-store.ts:248]: I think this is a duplicate line, this seems to cause a bunch of tests to fail on my machine

File: src/test/e2e/services/project-service.e2e.test.ts [hunk 2]
```diff
@@ -500,6 +500,44 @@ test('A newly created project only gets connected to enabled environments', asyn
...
+        name: second,
+        type: 'test',
+        sortOrder: 2,
+    });
+
+    await projectService.createProject(project, user);
+    const connectedEnvs =
+        await db.stores.projectStore.getEnvironmentsForProject(project.id);
+    expect(connectedEnvs).toHaveLength(4);
+    expect(connectedEnvs).toEqual([first, second, third, fourth]);
```
Comment (sighphyre) [inline src/test/e2e/services/project-service.e2e.test.ts:538]: You'll need to add the default environment here as well, that gets created by default and not having it in this list causes this to fail locally

**Review by sighphyre — CHANGES_REQUESTED** [review]
I don't necessarily have a problem with this change but a join and a sort is a bit of a computational cost on the db side. Should be a cold path so this doesn't seem like a problem but I think it's probably worth unpacking a little bit what the pain is on the front end side. I'm happy to approve but I think a third set of eyes would be sensible.

As an aside, I don't think the test that was added here actually ran on the build, I see 868 total on the build output but 869 total on my machine. Pretty sure that's nothing to do with this PR but it's definitely worth understanding

File: src/lib/db/project-store.ts [hunk 3]
```diff
@@ -238,7 +238,15 @@ class ProjectStore implements IProjectStore {
...
                 project_id: id,
             })
-            .pluck('environment_name');
+            .innerJoin(
+                'environments',
+                'project_environments.environment_name',
+                'environments.name',
+            )
+            .orderBy('environments.sort_order', 'asc')
+            .orderBy('project_environments.environment_name', 'asc')
```
Comment (FredrikOseberg) [inline src/lib/db/project-store.ts:247]: Won't this second orderBy override the orderBy on sort order?

Comment (sighphyre) [reply src/lib/db/project-store.ts:247]: My timing is terrible but I just found the answer - this is Knex's way of sorting on multiple columns: https://github.com/knex/knex/issues/2874

**Review by sighphyre — APPROVED** [review]
Thanks, looks good

**Pull request merged**